core/conanfile.py

```diff
@@ -23,6 +23,11 @@ class EmbeddedPythonCore(ConanFile):
     default_options = {"openssl_variant": "lowercase"}
     exports_sources = "embedded_python_tools.py", "embedded_python.cmake"
 
+    def validate(self):
+        minimum_python = "3.9.8"
+        if self.pyversion < minimum_python:
+            raise ConanInvalidConfiguration(f"Minimum supported Python version is {minimum_python}")
```

Comment:
Part of the reason is in the commit and the other part I forgot to write down:
* I removed the build workaround for `clang >= 13 and python < 3.9.8` in the same commit.
* In general, it's good to define a minimum supported version since we don't want to maintain old workarounds (above) and it's hard to test new features with all possible version (e.g. zip imports have gotten a lot of support in the latest versions of Python and it's not worth supporting older ones). We didn't define a minimum before, but we definitely didn't support 2.x, or even anything older than 3.7.
* In our project, we already use 3.9 or newer (mostly 3.10).